### PR TITLE
feat(leadspace-with-search): introducing new pattern (behind feature flag)

### PR DESCRIPTION
**Pull request by guilhermelMoraes**
### Description

This pattern was first introduced on the Support page. Here are the [visual specs](https://app.abstract.com/projects/7f83c114-7627-4169-8af7-6a85e7147ea6/branches/master/commits/latest/files/01ee3bd8-c63d-4b3b-bddb-aabaddb5c31d/layers/A3E455FA-D017-4878-8078-A0E1D9B0BCD1?collectionId=c963cb53-73c2-42a5-af39-f5538e24c454&collectionLayerId=3bbd5a17-7d2f-464d-8840-4794846b1227).

### Changelog

**New**

- `LeadspaceWithSearch` pattern;

### CODE REVIEW

File: packages/react/src/components/LeadspaceWithSearch/LeadspaceWithSearch.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Search from 'carbon-components-react/lib/components/Search';
+import settings from 'carbon-components/es/globals/js/settings';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+const LeadspaceWithSearch = ({ heading, copy, searchProps }) => {
```
Comment (jeffchew) [inline packages/react/src/components/LeadspaceWithSearch/LeadspaceWithSearch.js:17]: @guilhermelMoraes can you add a feature flag around this?

File: packages/react/src/components/LeadspaceWithSearch/README.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+Add the following line in your `.env` file at the root of your project.
+[See more details](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/styles#usage).
+
+```
+  SASS_PATH=node_modules:src
+```
+
+> 💡 Don't forget to import the Leadspace With Search styles from
+> [@carbon/ibmdotcom-styles](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/master/packages/styles).
+
```
Comment (jeffchew) [inline packages/react/src/components/LeadspaceWithSearch/README.stories.mdx:67]: Also add documentation on what feature flag to use to use this feature.

File: packages/react/src/components/LeadspaceWithSearch/README.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+## Stable selectors
+
+| Name                        | Description                      |
+| --------------------------- | -------------------------------- |
+| `bx--leadspace-with-search` | Pattern (grid container)         |
+| `dds--leadspace__row`       | LeadSpace grid row               |
+| `dds--leadspace__content`   | LeadSpace content (grid column)  |
+| `dds--leadspace__heading`   | Leadspace heading                |
+| `dds--leadspace__copy`      | LeadSpace optional copy          |
+| `dds--leadspace__search`    | LeadSpace search input           |
```
Comment (jeffchew) [inline packages/react/src/components/LeadspaceWithSearch/README.stories.mdx:81]: wondering if this should have different stable selectors from the regular leadspace...

File: packages/react/src/components/LeadspaceWithSearch/__stories__/LeadspaceWithSearch.stories.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import LeadspaceWithSearch from '../LeadspaceWithSearch';
+import React from 'react';
+import readme from '../README.stories.mdx';
+import { text } from '@storybook/addon-knobs';
+
+export default {
```
Comment (jeffchew) [inline packages/react/src/components/LeadspaceWithSearch/__stories__/LeadspaceWithSearch.stories.js:13]: Make sure you wrap the story with the feature flag

File: packages/styles/scss/ibm-dotcom-styles.scss [hunk 5]
```diff
@@ -55,6 +55,7 @@
 @import 'components/simplebenefits/simplebenefits';
 @import 'components/tableofcontents/tableofcontents';
 @import 'components/video-player/video-player';
+@import 'components/leadspace-with-search/index';
 @import 'components/promo-banner/index';
```
Comment (jeffchew) [inline packages/styles/scss/ibm-dotcom-styles.scss:59]: Can you add these in alphabetical order?

**Review by jeffchew — APPROVED** [review]
Looks good to me, thank you @guilhermelMoraes !

**Comment by jeffchew** [comment]
@guilhermelMoraes can you update the snapshot file?

**Comment by guilhermelMoraes** [comment]
@jeffchew we still have [this problem](https://cognitive-app.slack.com/archives/GTKP57FMJ/p1603743381008600) with the snapshot update. The `Search` component keeps changing its id which causes the snapshot check to fail. 

Also, a user called mfearer94 requested to add the typeahead feature.

**Comment by kennylam** [comment]
> @jeffchew we still have [this problem](https://cognitive-app.slack.com/archives/GTKP57FMJ/p1603743381008600) with the snapshot update. The `Search` component keeps changing its id which causes the snapshot check to fail.

@guilhermelMoraes Looks like the `Search` component accepts an `id` as a prop and will only randomize if one isn't passed. Simple solution is to give the `Search` an ID, e.g. `${prefix}--leadspace-with-search__search`. Just tested locally and was able to update snapshot and pass `ci-check`.

**Pull request merged**